config/management/network-address-encryption/src/lib.rs

```diff
@@ -0,0 +1,184 @@
+// Copyright (c) The Libra Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+#![forbid(unsafe_code)]
+
+use libra_global_constants::VALIDATOR_NETWORK_ADDRESS_KEYS;
+use libra_network_address::{
+    encrypted::{EncNetworkAddress, Key, KeyVersion},
+    NetworkAddress,
+};
+use libra_secure_storage::{Error as StorageError, KVStorage, Storage, Value};
+use move_core_types::account_address::AccountAddress;
+use serde::{Deserialize, Serialize};
+use std::collections::HashMap;
+use thiserror::Error;
+
+#[derive(Debug, Error)]
+pub enum Error {
+    #[error("Unable to deserialize address for account {0}: {1}")]
+    AddressDeserialization(AccountAddress, String),
+    #[error("Unable to decrypt address for account {0}: {1}")]
+    DecryptionError(AccountAddress, String),
+    #[error("Failed (de)serializing validator_network_address_keys")]
+    LCSError(#[from] lcs::Error),
+    #[error("Failed reading validator_network_address_keys from storage")]
+    StorageError(#[from] StorageError),
+    #[error("The specified version does not exist in validator_network_address_keys: {0}")]
+    VersionNotFound(KeyVersion),
+}
+
+pub struct Encryptor {
+    storage: Storage,
+}
+
+impl Encryptor {
+    pub fn new(storage: Storage) -> Self {
+        Self { storage }
+    }
+
+    pub fn add_key(&mut self, version: KeyVersion, key: Key) -> Result<(), Error> {
+        let mut keys = self.read()?;
+        keys.keys.insert(version, key);
+        self.write(&keys)
+    }
+
+    pub fn set_current_version(&mut self, version: KeyVersion) -> Result<(), Error> {
+        let mut keys = self.read()?;
+        if keys.keys.get(&version).is_some() {
+            keys.current = version;
+            self.write(&keys)
+        } else {
+            Err(Error::VersionNotFound(version))
+        }
+    }
+
+    pub fn current_version(&self) -> Result<KeyVersion, Error> {
+        self.read().map(|keys| keys.current)
+    }
+
+    pub fn encrypt(
+        &self,
+        network_address: &NetworkAddress,
+        account: AccountAddress,
+        seq_num: u64,
+    ) -> Result<Vec<u8>, Error> {
+        let keys = self.read()?;
+        let key = keys
+            .keys
+            .get(&keys.current)
+            .ok_or_else(|| Error::VersionNotFound(keys.current))?;
+        let raw_addr = std::convert::TryFrom::<&NetworkAddress>::try_from(network_address)?;
```

Comment:
into

Reply:
interesting that this is outdated: O, but I avoided pulling in the use std::... :P